.github/workflows/validation.yml

```diff
@@ -18,10 +21,22 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v3
 
-      - name: Install Python Dependencies
-        uses: HassanAbouelela/actions/setup-python@setup-python_v1.4.0
+      - name: Setup poetry
+        id: poetry_setup
+        uses: ItsDrike/setup-poetry@v1
+        with:
+          python-version: 3.11
```

Comment:
Let's run it on 3.8, so type checker will not allow `|` in type aliases (I remember, this somewhere has already appeared). Although this is not the right place to fix it, maybe.

```py
MyAlias: TypeAlias = ClassOne | ClassTwo  # no error on python 3.10+, but on 3.9 and lower is
```

Reply:
That's probably not something that this PR should be addressing. 

Also, pyright doesn't run the code it analyzes with the installed python version, it only performs static analysis based on the source code text. Dropping the python version for the CI will not make pyright run for that python version, since the installed pyright version will still be the same, and it's analysis is done for python 3.11 by default.

If this is something we'd want to change, all we need is to change the python version pyright is running for in pyright settings.